crates/edr_provider/src/data.rs

```diff
@@ -225,6 +225,39 @@ impl ProviderData {
             .map_err(ProviderError::Blockchain)
     }
 
+    pub fn block_by_number(
```

Comment:
Is there a reason this can't be in the handler by calling `block_by_block_spec`?

As far as I can tell, all of the methods you need to call are public APIs.

Reply:
Sure, alternative would be to have this as a free-standing function in `requests/eth/blocks.rs`. Want me to move it there?

Reply:
There is only one place that uses this function, right? It also seems like you followed this pattern with your previous PR

Reply:
Feel free to do either or, whichever follows the pattern

Reply:
I'm on the fence about this, because it feels more like provider logic than rpc logic (the other free-standing functions in `requests/**/*.rs` implement rpc logic), but on the other hand it's only used in `requests/eth/blocks.rs`.

Reply:
If we need it somewhere else, we can always move it later.

Reply:
Ok, done in [73a47ab](https://github.com/NomicFoundation/hardhat/pull/4628/commits/73a47abde285b00f1aaed640699256fa34585fb8)